### ISSUE TITLE
java.lang.Throwable: Slow operations are prohibited on EDT

### ISSUE THREAD

**Issue by ninthspace** [laravel-idea/plugin#328]
I have this exception, when I open a project, with this configuration with v4.3.2.212

PhpStorm 2021.2 EAP
Build #PS-212.4416.15, built on June 24, 2021
PhpStorm EAP User
Expiration date: July 24, 2021
Runtime version: 11.0.11+9-b1504.3 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-77-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 24
Registry: debugger.watches.in.variables=false, run.processes.with.pty=TRUE
Non-Bundled Plugins: com.jetbrains.FontDiagnostics (1.0), org.intellij.plugins.postcss (212.4416.11), org.jetbrains.plugins.stylus (212.4416.11), com.mallowigi (37.0), com.kalessil.phpStorm.phpInspectionsEA (4.0.6.4), dev.nybroe.collector (0.3.2), ru.adelf.idea.dotenv (2021.2.212), com.laravel_idea.plugin (4.3.2.212)
Current Desktop: zorin:GNOME

```
java.lang.Throwable: Slow operations are prohibited on EDT. See SlowOperations.assertSlowOperationsAreAllowed javadoc.
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:182)
	at com.intellij.util.SlowOperations.assertSlowOperationsAreAllowed(SlowOperations.java:108)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:749)
	at com.intellij.util.indexing.FileBasedIndexEx.processExceptions(FileBasedIndexEx.java:244)
	at com.intellij.util.indexing.FileBasedIndexEx.processValueIterator(FileBasedIndexEx.java:349)
	at com.intellij.util.indexing.FileBasedIndexEx.processValuesInScope(FileBasedIndexEx.java:318)
	at com.intellij.util.indexing.FileBasedIndexEx.processValues(FileBasedIndexEx.java:223)
	at com.intellij.psi.search.FilenameIndex.getVirtualFilesByName(FilenameIndex.java:209)
	at com.intellij.psi.search.FilenameIndex.getVirtualFilesByName(FilenameIndex.java:56)
	at com.intellij.psi.search.FilenameIndex.getVirtualFilesByName(FilenameIndex.java:51)
	at com.laravel_idea.plugin.extension.customConfig.LaravelCustomConfig.c(LaravelCustomConfig.kt:24)
	at com.laravel_idea.plugin.extension.settings.LaravelIdeaSettings$checkLaravelInComposer$1.run(LaravelIdeaSettings.kt:287)
	at com.intellij.openapi.project.DumbServiceImpl.doRun(DumbServiceImpl.java:401)
	at com.intellij.openapi.project.DumbServiceImpl.updateFinished(DumbServiceImpl.java:393)
	at com.intellij.openapi.project.TrackedEdtActivityService$TrackedEdtActivity.run(TrackedEdtActivityService.java:66)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:214)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:196)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:794)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:350)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:133)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:189)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:881)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:750)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:825)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:794)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:482)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by adelf** [comment]
Hi, Chris. Do you have any “ide.json” files in the project? If yes, could you share their contents?

**Comment by adelf** [comment]
Btw, it might be an EAP version bug. I’ll check. Thanks for creating the issue.

**Comment by ninthspace** [comment]
No, I don't have any ide.json files.

**Comment by enigmatic-user** [comment]
Same here with PhpStorm 2021.2 EAP Build 212.4535.18 (same Laravel Idea version as above), no ide.json file.

I found https://youtrack.jetbrains.com/issue/WI-58231 and https://youtrack.jetbrains.com/issue/WI-58226, but they have been reported for PhpStorm 2021.1, and the problem with Laravel Idea only exist in PhpStorm 2021.2 EAP (well, in my case).

**Comment by adelf** [comment]
Youtrack is full of these exceptions. More than 50 since the last EAP: https://youtrack.jetbrains.com/issues?q=Slow%20operations%20are%20prohibited%20on%20EDT
I'm sure it's an EAP bug. Anyway, it happens only in the project opening(and after the plugin settings change) and Laravel Idea works as expected. But thanks for reporting! I'll keep an eye on this.

**Comment by adelf** [comment]
Fixed in 4.3.3 version. Thanks for posting this issue!
